cmd/buildah/images.go

```diff
@@ -477,23 +531,31 @@ func matchesID(imageID, argID string) bool {
 	return strings.HasPrefix(imageID, argID)
 }
 
-func matchesReference(name, argName string) bool {
+func matchesReference(imageName, argName string) bool {
 	if argName == "" {
 		return true
 	}
-	splitName := strings.Split(name, ":")
-	// If the arg contains a tag, we handle it differently than if it does not
+	if imageName == "" {
+		return false
+	}
+	named, err := reference.ParseNormalizedNamed(imageName)
```

Comment:
Non-blocking: The `reference.Named` (or `nil`) could be passed down by the caller who has already done the parsing, but this works well enough.